root/static/scripts/edit/externalLinks.js

```diff
@@ -241,39 +242,55 @@ export class ExternalLinksEditor
               error = '';
             } else if (!link.url) {
               error = l('Required field.');
+              errorTarget = URLCleanup.ERROR_TARGETS.URL;
             } else if (isNewOrChangedLink && !isValidURL(link.url)) {
               error = l('Enter a valid url e.g. "http://google.com/"');
+              errorTarget = URLCleanup.ERROR_TARGETS.URL;
             } else if (isNewOrChangedLink && isMusicBrainz(link.url)) {
               error = l(`Links to MusicBrainz URLs are not allowed.
                          Did you mean to paste something else?`);
+              errorTarget = URLCleanup.ERROR_TARGETS.URL;
             } else if (isNewOrChangedLink && isMalware(link.url)) {
               error = l(`Links to this website are not allowed
                          because it is known to host malware.`);
+              errorTarget = URLCleanup.ERROR_TARGETS.URL;
             } else if (isNewOrChangedLink && isShortened(link.url)) {
               error = l(`Please don’t enter bundled/shortened URLs,
                          enter the destination URL(s) instead.`);
+              errorTarget = URLCleanup.ERROR_TARGETS.URL;
             } else if (isNewOrChangedLink && isGoogleAmp(link.url)) {
               error = l(`Please don’t enter Google AMP links,
                          since they are effectively an extra redirect.
                          Enter the destination URL instead.`);
+              errorTarget = URLCleanup.ERROR_TARGETS.URL;
             } else if (!link.type) {
               error = l(`Please select a link type for the URL
                          you’ve entered.`);
+              errorTarget = URLCleanup.ERROR_TARGETS.RELATIONSHIP;
             } else if (
               linkType.deprecated && (isNewLink || linkTypeChanged)
             ) {
               error = l(`This relationship type is deprecated 
                          and should not be used.`);
+              errorTarget = URLCleanup.ERROR_TARGETS.RELATIONSHIP;
             } else if (
               (linksByTypeAndUrl[linkTypeAndUrlString(link)] || []).length > 1
             ) {
               error = l('This relationship already exists.');
+              errorTarget = URLCleanup.ERROR_TARGETS.RELATIONSHIP;
             } else if (isNewOrChangedLink && checker) {
               const check = checker(link.url);
               if (!check.result) {
-                error = check.error ||
-                  l(`This URL is not allowed for the selected link type, 
-                     or is incorrectly formatted.`);
+                errorTarget = check.target ||
+                  URLCleanup.ERROR_TARGETS.NONE;
+                if (errorTarget === URLCleanup.ERROR_TARGETS.URL) {
+                  error = l(`This URL is incorrectly formatted.`);
```

Comment:
As you mentioned in your PR, "incorrectly formatted" can be confusing since it kinda sounds like the user can fix it by editing the formatting, which often won't be the case (otherwise, we would do it ourselves with a cleanup). Not that I'm sure what would be better, though. @mwiencek?

Reply:
This is where the default case I mentioned should be used for ambiguous situations.

Reply:
Maybe something saying that the domain is valid, but the page it points to isn't?  What other situations do we show this in?

Reply:
This message has been changed back to the old one as for now.